src/bp/core/repositories/sessions.ts

```diff
@@ -105,10 +105,19 @@ export class KnexSessionRepository implements SessionRepository {
   }
 
   async getExpiredContextSessionIds(botId: string): Promise<string[]> {
-    return (await this.database
+    let query = this.database
       .knex(this.tableName)
       .where('botId', botId)
       .andWhere(this.database.knex.date.isBefore('context_expiry', new Date()))
+
+    // We only process expired context if there is actually a context
+    if (this.database.knex.isLite) {
+      query = query.andWhereRaw(this.database.knex.raw(`context <> '{}' `))
+    } else {
+      query = query.andWhereRaw(this.database.knex.raw(`context::text <> '{}'::text`))
```

Comment:
What do you think to use the standard sql syntax for type cast instead of pg shorthand ?
Maybe that's not necessary because of your comment

Reply:
I have no opinion on the subject, I just found that these were working correctly. SQLite allows a simple comparison as text, while Postgres doesn't handle that for json type. 

If you know a better formula, i'd be happy to change it